### PR TITLE
Remove topshelf

### PR DESCRIPTION
**Pull request by yvesgoeleven**

### CODE REVIEW

File: src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs [hunk 1]
```diff
@@ -19,10 +19,7 @@ public void Start(IEnumerable<EndpointToHost> toHost)
             {
                 try
                 {
-                    var processStartInfo = new ProcessStartInfo(service.EntryPoint,
-                        "/serviceName:\"" + service.EndpointName +
-                        "\" /displayName:\"" + service.EndpointName +
-                        "\" /description:\"" + service.EndpointName + "\"")
+                    var processStartInfo = new ProcessStartInfo(service.EntryPoint)
```
Comment (yvesgoeleven) [inline src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs]: I have second thoughts here, maybe we should still pass these in when people use a new host to run old endpoints?

File: src/NServiceBus.Hosting.Azure.HostProcess/Program.cs [hunk 2]
```diff
@@ -2,132 +2,40 @@
...
+            settings.ShadowCopyFiles = "false";
+            settings.AppDomainInitializerArguments = args;
+            settings.ConfigurationFile = endpointConfigurationFile;
+            var domain = AppDomain.CreateDomain(endpointId, null, settings);
+            var windowsHost = domain.CreateInstanceAndUnwrap<WindowsHost>(endpointConfigurationType);
+            
+            windowsHost.Start();
 
-            var cfg = RunnerConfigurator.New(x =>
+            WaitHandle.WaitAny(new WaitHandle[]
```
Comment (SeanFeldman) [inline src/NServiceBus.Hosting.Azure.HostProcess/Program.cs]: Why this is needed?

File: src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs]: Those settings strike as windows service registration parameters. What would it do for the normal process?

**Comment by SeanFeldman** [comment]
Needs to be rebased.

File: src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs (hunk 1 above)
Comment (yvesgoeleven) [inline src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs]: I don't know, but the fact that it is here does seem to indicate that the process would not start without them, needs to be validated

File: src/NServiceBus.Hosting.Azure.HostProcess/Program.cs (hunk 2 above)
Comment (yvesgoeleven) [inline src/NServiceBus.Hosting.Azure.HostProcess/Program.cs]: otherwise the process exits

Comment (SeanFeldman) [inline src/NServiceBus.Hosting.Azure.HostProcess/Program.cs]: So an infinite loop.

File: src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs]: I'm hesitating to pull the trigger prior to confirmation.
Is this something you can test locally with the bits? Is anything else is needed?

Comment (yvesgoeleven) [inline src/NServiceBus.Hosting.Azure/DynamicHost/DynamicEndpointRunner.cs]: yes this can be tested locally, I'll do it on friday

**Pull request merged**